hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -367,20 +395,16 @@ public boolean hasNext() {
     }
 
     @Override
-    public T next() {
-      return reader.next();
+    public U next() {
+      return recordSupplier.get();
```

Comment:
Just add another iterator, it does not deserve to add another per-row invocation.

Reply:
Makes sense, cleaning it up now